File: src/main/java/org/example/bookfd/repos/AllNamesBooksRepo.java
package org.example.bookfd.repos;

import org.example.bookfd.domain.AllNamesBooks;
import org.springframework.data.repository.CrudRepository;

import java.util.List;

public interface AllNamesBooksRepo extends CrudRepository<AllNamesBooks,Integer> {
    List<AllNamesBooks> findByNumclass(int numclass);
    AllNamesBooks findByNumclassAndNamebook(int num,String namebook);
    AllNamesBooks findById(int id);
}

File: src/main/java/org/example/bookfd/repos/AllNamesBooksRepo.kt
package org.example.bookfd.repos

import org.example.bookfd.domain.AllNamesBooks
import org.springframework.data.repository.CrudRepository

interface AllNamesBooksRepo : CrudRepository<AllNamesBooks?, Int?> {
    fun findByNumclass(numclass: Int): List<AllNamesBooks?>?
    fun findByNumclassAndNamebook(num: Int, namebook: String?): AllNamesBooks?
    fun findById(id: Int): AllNamesBooks?
}
